crates/meroctl/src/cli/app/list.rs

```diff
@@ -50,13 +50,19 @@ impl Report for ListApplicationsResponse {
 
 impl ListCommand {
     pub async fn run(self, environment: &Environment) -> EyreResult<()> {
-        let config = load_config(&environment.args.home, &environment.args.node_name).await?;
+        let connection = environment
+            .connection
+            .as_ref()
+            .ok_or_else(|| eyre!("No connection configured"))?;
+
+        let mut url = connection.api_url.clone();
+        url.set_path("admin-api/dev/applications");
 
         let response: ListApplicationsResponse = do_request(
             &Client::new(),
-            multiaddr_to_url(fetch_multiaddr(&config)?, "admin-api/dev/applications")?,
+            url,
             None::<()>,
-            &config.identity,
+            connection.auth_key.as_ref(),
```

Comment:
not to hold back this PR any further so it certainly doesn't have to happen in this PR

but with the new design, it's very evident that there's still lots to improve upon with respect to the complexity of making requests

Now that we have a struct for connection info, we might as well keep the `Client` inside it

so at the end of the day, it becomes something like this:

```rust
environment
    .connection
    .get("admin-api/dev/applications", json!({}))
    .await?;
```

which significantly reduces the complexity and boilerplate needed to do really simple things

Reply:
Yeah, that makes more sense, but maybe we should get this merged first. If I see an issue opened to refactor the code on this section I'll open another PR for that.

Reply:
Sounds good! I'll file one.